conjure-java-jaxrs-client/src/test/java/com/palantir/conjure/java/client/jaxrs/TracerTest.java

```diff
@@ -60,26 +54,31 @@ public void before() {
     @Test
     public void testClientIsInstrumentedWithTracer() throws InterruptedException {
         server.enqueue(new MockResponse().setBody("\"server\""));
-        OpenSpan parentTrace = Tracer.startSpan("");
-        List<Map.Entry<SpanType, String>> observedSpans = Lists.newArrayList();
-        Tracer.subscribe(TracerTest.class.getName(),
-                span -> observedSpans.add(Maps.immutableEntry(span.type(), span.getOperation())));
 
+        String parentSpanId = Tracer.startSpan("").getSpanId();
         String traceId = Tracer.getTraceId();
+
+        List<String> spanIds = Lists.newArrayList();
+        List<Span> spans = Lists.newArrayList();
+        Tracer.subscribe(TracerTest.class.getName(), span -> {
+            spanIds.add(span.getSpanId());
+            spans.add(span);
+        });
+
         service.param("somevalue");
 
         Tracer.unsubscribe(TracerTest.class.getName());
-        assertThat(observedSpans, contains(
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-enqueue"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-enqueue"),
-                Maps.immutableEntry(SpanType.CLIENT_OUTGOING, "OkHttp: GET /{param}"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: dispatcher")));
+
+        assertThat(spans).containsExactly(
+                span(spans.get(0), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-enqueue"),
+                span(spans.get(1), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-run"),
```

Comment:
We could also just remove this span entirely and have the dispatcher span be directly under the directly. This would require adding a method to the `AsyncTracer` that only completes the async span:
```
public void completeSpan() {
    Trace originalTrace = Tracer.getAndClearTrace();
    Tracer.setTrace(deferredTrace);
    // Finish the enqueue span
    Tracer.fastCompleteSpan();
    Tracer.setTrace(originalTrace);
}
```

Reply:
Alternatively we could modify the `AsyncTracer` to allow for different operation names for the "execute" and "run" spans to improve the naming (something like "limiter-acquire" and "execute").

Reply:
I'd prefer to remove/deprecate AsyncTracer in favor of proper APIs to support what we want to measure. https://github.com/palantir/tracing-java/pull/124